src/expandableCalendar/calendarProvider.js

```diff
@@ -34,14 +42,86 @@ class CalendarProvider extends Component {
   };
 
   setDate = (date, updateSource) => {
-    this.setState({date, updateSource});
+    this.setState({date, updateSource}, () => {
+      this.animateTodayButton(date);
+    });
     _.invoke(this.props, 'onDateChanged', date, updateSource);
   }
+
+
+  getButtonIcon(date) {
+    const isPastDate = this.isPastDate(date);
+    return isPastDate ? require('../img/down.png') : require('../img/up.png');
+  }
+
+  isPastDate(date) {
+    const today = XDate();
+    const d = XDate(date);
+
+    if (today.getFullYear() > d.getFullYear()) {
+      return true;
+    }
+    if (today.getFullYear() === d.getFullYear()) {
+      if (today.getMonth() > d.getMonth()) {
+        return true;
+      }
+      if (today.getMonth() === d.getMonth()) {
+        if (today.getDate() > d.getDate()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  animateTodayButton(date) {
+    if (this.props.todayButton) {
+      this.setState({buttonIcon: this.getButtonIcon(date)});
+  
+      const today = XDate().toString('yyyy-MM-dd');
+      const isToday = today === date;
+      
+      Animated.spring(this.state.buttonY, {
+        toValue: isToday ? 65 : -65,
+        tension: 30,
+        friction: 8,
+        useNativeDriver: true
+      }).start();
+    }
+  }
+
+  onTodayPress = () => {
+    const today = XDate().toString('yyyy-MM-dd');
+    this.setState({date: today, updateSource: UPDATE_SOURCES.TODAY_PRESS});
+  }
+
+  renderTodayButton() {
+    const todayString = XDate.locales[XDate.defaultLocale].today || commons.todayString;
+    const today = todayString.charAt(0).toUpperCase() + todayString.slice(1);
+    
+    return (
+      <Animated.View 
+        style={{
+          position: 'absolute', 
```

Comment:
move this part of the style to a stylesheet

Reply:
Done.